content/en/blog/_posts/2022-10-31-forensic-container-checkpointing/index.md

```diff
@@ -0,0 +1,199 @@
+forensic container checkpointing as implemented in Kubernetes
+uses these existing CRIU integrations.
+
+[criu]: https://criu.org/
+
+## Why is it important?
+
+With the help of CRIU and the corresponding integrations it is possible to get
+all information and state about a running container on disk for later forensic
+analysis. Forensic analysis might be important to inspect a suspicious
+container without stopping or influencing it. If the container is really under
+attack, the attacker might detect attempts to inspect the container. Taking a
+checkpoint and analysing the container in a sandboxed environment offers the
+possibility to inspect the container without the original container and maybe
+attacker being aware of the inspection.
+
+In addition to the forensic container checkpointing use case, it is also
+possible to migrate a container from one node to another node without loosing
+the internal state. Especially for stateful containers with long initialization
+times restoring from a checkpoint might save time after a reboot or enable much
+faster startup times.
+
+## How do I use container checkpointing?
+
+The feature is behind a feature gate, so make sure to enable the
+`ContainerCheckpointRestore` gate before you can use the new feature.
+
+The runtime must also support forensic container checkpointing:
+
+* containerd: support is currently under discussion. See containerd
+  pull request [#6965][containerd-checkpoint-restore-pr] for more details.
+
+* CRI-O: v1.25 has support for forensic container checkpointing.
+
+[containerd-checkpoint-restore-pr]: https://github.com/containerd/containerd/pull/6965
+
+### Usage example with CRI-O
+
+To use forensic container checkpointing in combination with CRI-O, CRI-O needs
+to be started with the command-line options `--enable-criu-support=true
+--drop-infra-ctr=false` and Kubernetes needs to be started with the
+`ContainerCheckpointRestore` feature gate enabled. As the checkpointing
+functionality is provided by CRIU it is also necessary to install CRIU.
+Usually runc or crun depend on CRIU and therefore it is installed automatically.
+
+Once containers and pods are running it is possible to create a checkpoint.
+Checkpointing is currently only exposed on the **kubelet** level. To checkpoint
+a container, you can run `curl` on the node where that container is running,
+and trigger a checkpoint:
+
+```shell
+curl -X POST "https://localhost:10250/checkpoint/namespace/podId/container"
+```
+
+For a container named *counter* in a pod named *counters* in a namespace named
+*default* the **kubelet** API endpoint is reachable at:
+
+```shell
+curl -X POST "https://localhost:10250/checkpoint/default/counters/counter"
+```
+
+For completeness the following `curl` command-line options are necessary to
+have `curl` trust the kubelet's CA certificate:
+
+```shell
+-k --cert /var/run/kubernetes/client-admin.crt --key /var/run/kubernetes/client-admin.key
```

Comment:
`-k` does not _trust_ the cert, it instead **skips** a security check. Omit `-k` here (and add other options around the CA certificate), or reword to explain what's going on.